### ISSUE TITLE
POM-based configuration properties not used

### ISSUE THREAD

**Issue by lancearlaus** [deliveredtechnologies/terraform-maven#153]
# Summary
Plugin does not support documented configuration properties when specified as part of a POM-based configuration block. Developer experience also reduced since IDE flags what should be valid configuration properties as invalid.

# Steps to reproduce
* Maven configuration file with a configured plugin execution
* Documented configuration properties, other than tfRootDir and/or artifact, are not passed to plugin execution

## Sample Configuration Snippet
```
<execution>
  <id>someId</id>
  <phase>pre-integration-test</phase>
  <configuration>
    <!-- The following configuration is ignored -->
    <tfVars>PROJECT_USER=${user.name},PROJECT_NAME=${project.name}</tfVars>
  </configuration>
  <goals>
    <goal>apply</goal>
  </goals>
</execution>
```

# Analysis

* The API has the options enumerated, but the plugin's Mojo classes don't have typical annotated properties that allow Maven and associated tooling to introspect and pick up the configuration properties.
* When instantiating the plugin classes, system properties are passed in which doesn't allow for execution-specific configuration
* Testing of configuration options appears limited to API level, not POM-based verification
* Maven appears to require configuration parameter metadata for correct operation (just adding what should be the appropriate configuration block to the POM doesn't work since the values are ignored)

# Suggested Fix
* Add the appropriate configuration properties and/or metadata
* Bonus: Annotating the plugin's goals with sensible default phases would reduce boilerplate configuration required to use the plugin

**Comment by Clayton7510** [comment]
Thanks for the detailed description. 0.9-SNAPSHOT contains the Mojo Parameter annotations so that they can be used in the configuration blocks in the POM.

I have been putting some thought into sensible default phases. I'll update 0.9-SNAPSHOT shortly with those and you can let me know what you think. (y)

**Comment by lancearlaus** [comment]
Thanks for the attention on the issue. I commented on the commit. Hopefully it makes sense. I can work on a pull request aligned with my comments, if you're interested. Just let me know.

**Comment by Clayton7510** [comment]
I appreciate the feedback. Good points made. I updated the goals to include properties that have specific types, when applicable. I left tfRootDir since it's usage was overloaded. It could be a file, but it could also be just the name of a module that by convention would match up to src/main/tf/{module name}. I also left it so that the TerraformOperation subclasses could accept either String properties or properties of the specific type since Properties are a bit easier to work with using Strings and there was sort of an implicit contract there - didn't want to break working functionality.

If I hadn't already started working on this, I would have definitely loved a PR. My experience with other open source projects is that most folks aren't super eager to make contributions. So, before you commented on the commit, I figured that the work was on me ;) In the future, if you put in the issue that you'd like to work on it and submit a PR, I would hold off in anticipation of the contribution.

Anyway, I'll push the update to the 0.9-SNAPSHOT, spend a couple of days playing with it and if it seems to work well, then I'll merge it Sun/Mon. Also, if you play with it and find anything not working as expected, additional feedback is always welcome.
